### PR TITLE
[videodb] fetch season year based on first aired episode

### PR DESCRIPTION
**Pull request by mkortstiege**
This adds episode based premiered/first aired date to season view. Tvshow based premiered date is used as fallback. Request @ http://forum.kodi.tv/showthread.php?tid=226638.

Having premiered date for season based on the tvshow makes no sense actually ;) As this is a pretty minor change which IMO adjusts the false logic used in there, I'd even consider this as a fix for Isengard. No biggie if it has to wait for J***.

@Montellese for review please.

### CODE REVIEW

**Comment by mkortstiege** [comment]
jenkins build this please

File: xbmc/video/VideoDatabase.cpp [hunk 1]
```diff
@@ -6031,9 +6032,18 @@ bool CVideoDatabase::GetSeasonsByWhere(const std::string& strBaseDir, const Filt
...
         pItem->GetVideoInfoTag()->m_strPlot = m_pDS->fv(VIDEODB_ID_SEASON_TVSHOW_PLOT).get_asString();
+
+        // tvshow premiered date - used as fallback in case no episode based
+        // premiered/first aired is available for the given season
         pItem->GetVideoInfoTag()->m_premiered.SetFromDBDate(m_pDS->fv(VIDEODB_ID_SEASON_TVSHOW_PREMIERED).get_asString());
         if (pItem->GetVideoInfoTag()->m_premiered.IsValid())
           pItem->GetVideoInfoTag()->m_iYear = pItem->GetVideoInfoTag()->m_premiered.GetYear();
+
+        // season premiered date - based on first episode airdate associated to the season
+        pItem->GetVideoInfoTag()->m_firstAired.SetFromDBDate(m_pDS->fv("aired").get_asString());
```
Comment (Montellese) [inline xbmc/video/VideoDatabase.cpp]: Please use an enum constant instead of `"aired"` like everywhere else.

File: xbmc/video/VideoDatabase.cpp [hunk 2]
```diff
@@ -6031,9 +6032,18 @@ bool CVideoDatabase::GetSeasonsByWhere(const std::string& strBaseDir, const Filt
...
+        if (pItem->GetVideoInfoTag()->m_firstAired.IsValid())
+          pItem->GetVideoInfoTag()->m_iYear = pItem->GetVideoInfoTag()->m_firstAired.GetYear();
```
Comment (Montellese) [inline xbmc/video/VideoDatabase.cpp]: Maybe move the logic to set the year together to make it more obvious which has preference over the other like
```
if (pItem->GetVideoInfoTag()->m_firstAired.IsValid())
  pItem->GetVideoInfoTag()->m_iYear = pItem->GetVideoInfoTag()->m_firstAired.GetYear();
else if (pItem->GetVideoInfoTag()->m_premiered.IsValid())
  pItem->GetVideoInfoTag()->m_iYear = pItem->GetVideoInfoTag()->m_premiered.GetYear();
```

**Comment by Montellese** [comment]
Looks good apart from the two comments.

**Comment by mkortstiege** [comment]
Updated according your comments

**Comment by Montellese** [comment]
Great thanks.

**Comment by mkortstiege** [comment]
jenkins build this please

**Comment by mkortstiege** [comment]
@Montellese, do we want this in for Isengard or hold back for J***?

**Comment by Montellese** [comment]
IMO it's not that big of a deal so I'd wait for J*** to avoid bumping the videodb version at this point in the release.

**Comment by mkortstiege** [comment]
Agreed.

**Comment by mkortstiege** [comment]
Rebased. jenkins build this please.

**Comment by mkortstiege** [comment]
Build errors are unrelated.

**Pull request merged**